### PR TITLE
Fix issue #11, enable integer redshift inputs to preprocessing 

### LINKED ISSUE #11: Tensorflow issues with integer inputs
A few users have reported issues with integer redshift inputs not being handled properly by tensorflow because they are not cast to float32s before being passed on. This is an easy fix where all training data/labels and test data/labels are properly cast to float32s in the `preprocess.py` file.

### PR DESCRIPTION
**Pull request by ThomasGesseyJones**
resolves #11

### CODE REVIEW

**Comment by htjb** [comment]
Looks great @ThomasGesseyJones, thanks for sorting this! Could you just remove python 3.6 from the `CI.yml` in `.github/workflows/`. It seems this was deprecated at the beginning of this year (https://devguide.python.org/versions/).

Feel free to squash and merge once the tests are passing.
